### PR TITLE
move send unhealthy endpoint flag to atomic bool

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixes https://github.com/istio/istio/issues/39925

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@howardjohn PTAL if this makes sense as a pattern. If it does, I will move rest of the feature flag in separate PR.

File: pilot/pkg/features/pilot.go [hunk 1]
```diff
@@ -104,13 +105,13 @@ var (
 			" EDS pushes may be delayed, but there will be fewer pushes. By default this is enabled",
 	).Get()
 
-	SendUnhealthyEndpoints = env.RegisterBoolVar(
+	SendUnhealthyEndpoints = atomic.NewBool(env.RegisterBoolVar(
```
Comment (howardjohn) [inline pilot/pkg/features/pilot.go:108]: I do wonder if we should change the `env` package -- possibly at the same time as making it use generics -- to make it so everything is atomic?

This problem is universal, just happens that one test triggers it most often

**Review by howardjohn — APPROVED** [review]

File: pilot/pkg/features/pilot.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/features/pilot.go:108]: > I do wonder if we should change the env package -- possibly at the same time as making it use generics -- to make it so everything is atomic?

Yes.
Should I change others to also this model for now and update env package later when we have generics?

**Comment by howardjohn** [comment]
Yeah lets wait

**Pull request merged**